pkg/types/conversion/tfjson/tfjson.go

```diff
@@ -147,6 +156,36 @@ func tfJSONBlockTypeToV2Schema(nb *tfjson.SchemaBlockType) *schemav2.Schema { //
 	return v2sch
 }
 
+func schemaV2TypeFromNestedType(ntyp *tfjson.SchemaNestedAttributeType, schema *schemav2.Schema) error {
+	switch ntyp.NestingMode {
+	case tfjson.SchemaNestingModeSingle:
+		schema.Type = schemav2.TypeList
+		schema.MinItems = 0
+		schema.MaxItems = 1
+	case tfjson.SchemaNestingModeGroup:
+		schema.Type = schemav2.TypeList
+		schema.MinItems = 1
+		schema.MaxItems = 1
+	case tfjson.SchemaNestingModeList:
+		schema.Type = schemav2.TypeList
+	case tfjson.SchemaNestingModeSet:
+		schema.Type = schemav2.TypeSet
+	case tfjson.SchemaNestingModeMap:
+		schema.Type = schemav2.TypeMap
+	default:
+		return errors.Errorf("unknown nesting mode: %s", ntyp.NestingMode)
```

Comment:
Especially, if the nesting mode string turns out to be an empty string:
nit:
```suggestion
		return errors.Errorf("unknown nesting mode: %q", ntyp.NestingMode)
```